pyledger/ledger_engine.py

```
@@ -2153,3 +2157,96 @@ def _report_column(self, row, accounts, prune_level=2) -> pd.DataFrame:
+        profit_centers: str | None = None,
+        root_folder: str | None = None,
+        output: Literal["typst", "dataframe"] = "typst",
+        format_number: Callable[[float], str] | None = None,
+    ) -> dict[str, str | pd.DataFrame]:
+        """
+        Generate formatted tables with transaction history for each account.
+
+        Retrieves the transaction history For each account over the specified
+        period, formats the data, and returns a table for each account either
+        as Typst-formatted strings or DataFrames.
+
+        Args:
+            period (str): Period or date range for transactions. Accepts values such as
+                a year ("2024"), month ("2024-01"), quarter ("2024-Q1"), or a full
+                start-end span. See `parse_date_span` for accepted formats.
+            config (pd.DataFrame): Labels, widths, and alignments of columns to include.
+            profit_centers (str | None, optional): Optional filter to include only
+                transactions for the specified profit center(s).
+            root_folder (str | None, optional): If provided, values in the 'document'
+                column will be rendered as links in Typst outout, concatenating this
+                base path with the value of the document column.
+            output (Literal["typst", "dataframe"]): Determines return type: either
+                rendered Typst strings or DataFrames. Defaults to "typst".
+            format_number (Callable[[float], str], optional): Function to format numeric values.
+                If None, a default formatter is used with number of decimal digits matching
+                the reporting currency's precision.
+
+        Returns:
+            dict[str, str | pd.DataFrame]: A mapping from account numbers to either Typst
+            table strings or formatted DataFrames, depending on `output`.
+        """
+        config = enforce_schema(config, ACCOUNT_SHEET_REPORT_CONFIG_SCHEMA)
```

Please raise an error if users provides config columns that do not exists in account history and